PL2/PL2-gui/src/main/java/nl/tudelft/pl2016gr2/gui/model/PhylogeneticTreeNode.java

```diff
@@ -1,109 +1,240 @@
 package nl.tudelft.pl2016gr2.gui.model;
 
+import javafx.beans.property.BooleanProperty;
+import javafx.beans.property.SimpleBooleanProperty;
+import net.sourceforge.olduvai.treejuxtaposer.drawer.TreeNode;
+
 import java.util.ArrayList;
-import java.util.Objects;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
 
 /**
- * This class is an adapter class which maps the methods of the
- * net.sourceforge.olduvai.treejuxtaposer.drawer.TreeNode class to the interface which is needed by
- * our application.
+ * This class is a storage class for a phylogenetic tree node. It retrieves and stores the needed
+ * values of the newick parser {@link TreeNode}.
  *
  * @author Faris
  */
-public class PhylogeneticTreeNode implements IPhylogeneticTreeNode {
+public class PhylogeneticTreeNode implements IPhylogeneticTreeNode, Iterable<PhylogeneticTreeNode> {
+
+  private final String label;
+  private final float weight;
+  private final PhylogeneticTreeNode[] children;
+  private final PhylogeneticTreeNode parent;
```

Comment:
It is also used in the `hasParent()` method. The UI only stores the current root node (which is the root of the part of the tree which is currently shown on the screen), so to zoom out the parent is needed.
The `setDrawnInBottom` and `setDrawnInTop` also use the parent: to propagate the value of drawnInTop and drawnInBottom to the parent class whenever it changes (ex. when both children of a node are drawn in top, the node must be marked as drawn in top). This could be done with listeners, but I don't really see an advantage of listeners for this case.